src/hotspot/share/opto/compile.cpp

```diff
@@ -4948,6 +4949,34 @@ bool Compile::coarsened_locks_consistent() {
   return true;
 }
 
+// Mark locking regions (identified by BoxLockNode) as coarsened
+// if locks coarsening optimization removed lock/unlock from them.
+// Such regions become unbalanced and we can't execute other
+// locks elimination optimization on them.
```

Comment:
The comment suggests you mark boxes as "coarsened" if ...
But really you mark them as "unbalanced", right? Because you have the condition that the locks have to already be "coarsened".

Reply:
I updated comment